core/src/main/java/org/eqasim/core/simulation/analysis/EqasimAnalysisModule.java

```diff
@@ -17,8 +26,35 @@ public class EqasimAnalysisModule extends AbstractModule {
 	@Override
 	public void install() {
 		addControlerListenerBinding().to(AnalysisOutputListener.class);
-		bind(DefaultPersonAnalysisFilter.class);
-		bind(PersonAnalysisFilter.class).to(DefaultPersonAnalysisFilter.class);
+		if(getConfig().getModules().containsKey(MultiModeDrtConfigGroup.GROUP_NAME)) {
+			bind(VehicleRegistry.class).asEagerSingleton();
+			addEventHandlerBinding().to(VehicleRegistry.class);
+			//Using a static method as a provider causes it to be called even when the bind is not performed
+			//So to avoid the potential problems, we use the provider below
+			addControlerListenerBinding().toProvider(new Provider<>() {
```

Comment:
Please rather put a @Provides @Singleton further below and just add DrtAnalysisListener as a listener binding

Reply:
As explained in the comments, when I tried to use @provides and @singleton it broke compatibility with non-drt simulations. Guice still wanted to run the provider even if the bind never happens, but then crashed when the MultiModeDrtConfigGroup was not found.

Reply:
Ok I think It's resolved now. 
See below